### PR TITLE
feat: Unify different ma-button and ma-link sizes

### PR DESCRIPTION
**Pull request by carlotamolintr**
We have MaButtons of different sizes in different apps. There are problems with the font size, the paddings and the horizontal styles. This PR fix the issue :)

I just added the width property to set full width in case we want the button to be in full width. By default the prop is set text-width.

### CODE REVIEW

File: src/components/MaLink/MaLink.vue [hunk 1]
```diff
@@ -1,11 +1,14 @@
 <template>
-  <a :class="`ma-link--${category} ma-link`" :href="href" :rel="computedRel">
+  <a :class="[getClasses, 'ma-link']" :href="href" :rel="computedRel">
     <slot />
   </a>
 </template>
 
 <script lang="ts">
 import Vue, { PropOptions } from 'vue'
+import { Prop } from 'vue/types/options'
```
Comment (alex-castelo) [inline src/components/MaLink/MaLink.vue:9]: I think that with `PropOptions`should be enough. This, is it being used?

File: src/components/MaButton/MaButton.vue [hunk 2]
```diff
@@ -72,6 +72,15 @@ export default Vue.extend({
...
      */
+
+    /**
+     * Sets the button font-size style
+     */
+    size: {
+      type: String,
+      default: 'small',
+      validator: (val) => ['small', 'large'].includes(val),
+    },
```
Comment (alex-castelo) [inline src/components/MaButton/MaButton.vue:83]:
```suggestion
    size: {
      type: String,
      default: 'small',
      validator: (val) => ['small', 'large'].includes(val),
    } as PropOptions<ButtonSize>,
```

In order to keep this TS inference, maybe we need to create this type on top of compoment. What do you think about it? :)

**Review by alex-castelo — APPROVED** [review]
I left a couple of comments! Good job Carlota 🚀

File: src/components/MaButton/MaButton.vue (hunk 2 above)
Comment (carlotamolintr) [reply src/components/MaButton/MaButton.vue:83]: Yes! I just forgot it... it didn't give me an error... weird...

File: src/components/MaLink/MaLink.vue (hunk 1 above)
Comment (carlotamolintr) [reply src/components/MaLink/MaLink.vue:9]: This was automatically imported I guess 🤔. I will remove it!

File: src/components/MaButton/MaButton.vue (hunk 2 above)
Comment (alex-castelo) [reply src/components/MaButton/MaButton.vue:83]: Maybe not an error. But if we want the IntelliSense features for autocomplete. I think we need it

**Review by sergigraciaj — APPROVED** [review]

**Review by xevops — APPROVED** [review]

File: src/components/MaButton/MaButton.vue [hunk 3]
```diff
@@ -111,7 +111,7 @@ export default Vue.extend({
         'ma-button--rounded': this.rounded,
         [`ma-button--${this.category}`]: this.category,
         'ma-button--loading': this.loading,
-        [`ma-button--${this.size}`]: this.size,
+        [`ma-button--${this.size}`]: this.size && !this.rounded,
```
[the remark this answers is not in the document]
Comment (cillascillas) [reply src/components/MaButton/MaButton.vue]: Same as [here](https://github.com/holaluz/margarita/pull/535#discussion_r896636563)

**Review by cillascillas — APPROVED** [review]

**Pull request merged**